root/types.js

```diff
@@ -1,24 +1,72 @@
-// @flow
-// Copyright (C) 2017 MetaBrainz Foundation
-//
-// This file is part of MusicBrainz, the open internet music database,
-// and is licensed under the GPL version 2, or (at your option) any
-// later version: http://www.gnu.org/licenses/gpl-2.0.txt
+/*
+ * @flow
+ * Copyright (C) 2017 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
 
 
-// Types are in alphabetical order.
-//
-// The definitions in this file are intended to model the output of the
-// TO_JSON methods under lib/MusicBrainz/Server/Entity/; those are precisely
-// how data is serialized for us.
+/*
+ * Types are in alphabetical order.
+ *
+ * The definitions in this file are intended to model the output of the
+ * TO_JSON methods under lib/MusicBrainz/Server/Entity/, those are precisely
+ * how data is serialized for us.
+ */
+
+declare type AreaT =
+  & CoreEntityT
+  & TypeRoleT<AreaTypeT>
+  & {|
+      +containment?: $ReadOnlyArray<AreaT>,
+      +entityType: 'area',
+    |};
+
+export opaque type AreaTypeT: OptionTreeT = OptionTreeT;
```

Comment:
Opaque types are documented at https://flow.org/en/docs/types/opaque-types/

They need `opaque` because otherwise e.g. you could pass an `AreaTypeT` where an `ArtistTypeT` is expected, since they are both aliases to `OptionTreeT`. The `opaque` hides the underlying type from code outside the file.